packages/expo-video/android/src/main/java/expo/modules/video/records/VideoSource.kt

```diff
@@ -57,4 +63,31 @@ class VideoSource(
       )
     }
     .build()
+
+  // Using expo-asset to generate a local asset URI returns a resource name for android production builds
+  // we have to get the raw resource URI to play the video
+  @SuppressLint("DiscouragedApi") // AFAIK, in this case, there's no other way to get the resource URI
+  private fun parseLocalAssetId(uri: Uri?, context: Context): Uri? {
+    if (uri == null || uri.scheme != null) {
+      return uri
+    }
+    try {
+      val resourceId: Int = context.resources.getIdentifier(
+        uri.toString(),
+        "raw",
+        context.packageName
+      )
+      val parsedUri = Uri.Builder()
+        .scheme(ContentResolver.SCHEME_ANDROID_RESOURCE)
+        .appendPath(resourceId.toString())
+        .build()
+      val dataSpec = DataSpec(parsedUri)
+      val rawResourceDataSource = RawResourceDataSource(context)
+      rawResourceDataSource.open(dataSpec)
+      return rawResourceDataSource.uri
+    } catch (e: RawResourceDataSource.RawResourceDataSourceException) {
+      Log.e("ExpoVideo", "Error parsing local asset id, falling back to original uri", e)
+      return uri
+    }
+  }
```

Comment:
I don't like this approach. In expo-image we're using `resolveAssetSource` which do this on js side. Can we use the same function here? 

Reply:
We are doing the same thing in `expo-image` https://github.com/expo/expo/blob/91170d8cd8da436ad268b3446fbd897a6e423f59/packages/expo-image/android/src/main/java/expo/modules/image/ResourceIdHelper.kt#L11-L45. Using `resolveAssetSource` instead of depending on `expo-asset` is a great idea though, now we can have integrated `require` support 😄